doc/nrf/gs_modifying.rst

```diff
@@ -14,25 +14,49 @@ After programming and testing an application, you probably want to make some mod
 Adding files and changing compiler settings
 *******************************************
 
-All files that your application uses must be specified in the :file:`CMakeLists.txt` file.
-By default, most samples include only the main application file :file:`src/main.c`.
-This means that you must add all other files that you are using.
+The |NCS| build system is based on Zephyr, that has build system based on `CMake`_.
+For more information see :ref:`application`.
 
-You can also configure compiler options, application defines, or include directories, or set :ref:`build types <gs_modifying_build_types>` in :file:`CMakeLists.txt`.
+The application or sample in the |NCS| is a `CMake`_ project.
+As such it controls the configuration and build process of both itself, Zephyr and any sourced libraries.
+The application's :file:`CMakeLists.txt` file is the main `CMake`_ project file and the source of the build process configuration.
+
+The Zephyr creates a `CMake`_ component that must be imported by the application in its :file:`CMakeLists.txt` file.
```

Comment:
Maybe worth mentioning that this is usually achieved by `find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})` command in CMake so a user knows which instruction this text refers to?